src/err.rs

```diff
@@ -56,7 +56,29 @@ pub struct PyErr {
 pub type PyResult<T> = Result<T, PyErr>;
 
 /// Marker type that indicates an error while downcasting
-pub struct PyDowncastError;
+pub struct PyDowncastError {
+    from: Option<String>,
+    to: Option<String>,
+}
```

Comment:
I have a concern that creating two strings for most conversion errors may lead to a noticeable performance overhead.

How about this alternative:

```
pub struct PyDowncastError<'a> {
    from: &'a PyAny,
    to: Cow<'a, str>,
}
```

I suggest `from` becomes `&'a PyAny` because I think I'd rather see `from.repr()` rather than just the source type. That'd make it easier for me to debug out exactly what value is failing to convert.

And `from` as `Cow<'a, str>` just so we can avoid copying the string when we don't have to :)